src/renderer/src/lib/locale/en/errors.json

```
@@ -3,22 +3,24 @@
 	"appNameExistsError": "An app with the same name is already installed. Please choose a different name",
 	"appStoreClientNotInitializedError": "AppStoreClient not initialized",
 	"checkInitializedKeystoreError": "Error checking initialized keystore",
-	"duplicatePubKeyError": "An app with the same public key is already installed. This is not allowed due to security reasons.",
 	"devHubClientNotInitializedError": "DevHubClient not initialized",
+	"duplicatePubKeyError": "An app with the same public key is already installed. This is not allowed due to security reasons.",
+	"failedForAllAvailableHosts": "Downloading the app failed for all available peers. This typically happens due to none of the online peers having the app available themselves. Retrying again later can resolve this if new peers come online.",
 	"failedToCreateSymlinkedLairDirectoryError": "Failed to create symlinked lair directory",
 	"fileUndefinedError": "File undefined",
 	"initializeLairKeystoreError": "Error initializing lair keystore",
 	"launchLairKeystoreError": "Error launching lair keystore",
 	"missingBinaries": "There are missing holochain or lair keystore binaries",
 	"noAppPortError": "Can't find app port",
 	"noAppstoreAuthenticationTokenFound": "No authentication token found for appstore",
-	"noAvailablePeerHostsError": "No available peer hosts found.",
+	"noAvailablePeerHostsError": "It looks like no peer is online from which you can download this app. Please try again later.",
```

I don't like vagueness in a binary context =D
```suggestion
	"noAvailablePeerHostsError": "No peer is online from which you can download this app. Please try again later.",
```

---

Agreed. The term "online" is not even well defined though but that's another discussion...

---

https://github.com/holochain/launcher/pull/248/commits/2fd5b772a41e8b3ecc4688032eba63bc6ef868aa

---

You could change it to "available".